fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineShortcuts.kt

```diff
@@ -0,0 +1,273 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.search.SearchEngine
+import mozilla.components.browser.state.state.SearchState
+import mozilla.components.browser.state.state.availableSearchEngines
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Top-level UI for search shortcuts settings.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun SearchEngineShortcuts(
+    categoryTitle: String,
+    store: BrowserStore,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+    onAddEngineClicked: () -> Unit,
+) {
+    val searchState = store.observeAsComposableState { it.search }.value ?: SearchState()
+    val searchEngines = with(searchState) {
+        regionSearchEngines + additionalSearchEngines + availableSearchEngines + customSearchEngines
+    }
+    val disabledShortcutsIds = searchState.disabledSearchEngineShortcutsIds
+
+    LazyColumn(
+        modifier = Modifier
+            .background(color = FirefoxTheme.colors.layer1)
+            .fillMaxSize(),
+    ) {
+        item {
+            Title(title = categoryTitle)
+
+            Spacer(modifier = Modifier.height(12.dp))
+        }
+
+        items(
+            items = searchEngines,
+            key = { engine -> engine.id },
+        ) {
+            SearchItem(
+                engine = it,
+                name = it.name,
+                isEnabled = !disabledShortcutsIds.contains(it.id),
+                onCheckboxClicked = onCheckboxClicked,
+                onEditEngineClicked = onEditEngineClicked,
+                onDeleteEngineClicked = onDeleteEngineClicked,
+            )
+        }
+
+        item {
+            AddEngineButton(onAddEngineClicked = onAddEngineClicked)
+        }
+    }
+}
+
+@Composable
+private fun Title(title: String) {
+    Box(
+        modifier = Modifier
+            .height(48.dp)
+            .fillMaxWidth(),
+        contentAlignment = Alignment.CenterStart,
+    ) {
+        Text(
+            text = title,
+            color = FirefoxTheme.colors.textAccent,
+            fontWeight = FontWeight.W400,
+            modifier = Modifier.padding(horizontal = 16.dp),
+            style = FirefoxTheme.typography.headline8,
+        )
+    }
+}
+
+@Suppress("LongParameterList", "LongMethod")
+@Composable
+private fun SearchItem(
+    engine: SearchEngine,
+    name: String,
+    isEnabled: Boolean,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+) {
+    val isMenuExpanded: MutableState<Boolean> = remember { mutableStateOf(false) }
+
+    Row(
```

Comment:
Or just extend the existing [`RadioButton`](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/RadioButton.kt). If there isn't a figma, we can shoot Crystal a quick DM to figure this one out. 

I would definitely encourage breaking this PR down into smaller chunks though. It'll help with the reviewing, obviously, but it will also make any round of discussion and iteration that much quicker.

Reply:
created a [follow up](https://bugzilla.mozilla.org/show_bug.cgi?id=1837113)